src/CodeGenerator/src/Generator/RequestSerializer/JsonSerializer.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AsyncAws\CodeGenerator\Generator\RequestSerializer;
+
+use AsyncAws\CodeGenerator\Definition\ListShape;
+use AsyncAws\CodeGenerator\Definition\MapShape;
+use AsyncAws\CodeGenerator\Definition\Member;
+use AsyncAws\CodeGenerator\Definition\Operation;
+use AsyncAws\CodeGenerator\Definition\Shape;
+use AsyncAws\CodeGenerator\Definition\ShapeWithMembers;
+use AsyncAws\CodeGenerator\Definition\StructureMember;
+use AsyncAws\CodeGenerator\Definition\StructureShape;
+use AsyncAws\CodeGenerator\Generator\Naming\NamespaceRegistry;
+
+/**
+ * Serialize a request body to a nice nested array.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ *
+ * @internal
+ */
+class JsonSerializer implements Serializer
```

Comment:
I copied the RestJsonSerializer. We can probably merge them later. But I'll do that in a separate PR.

Reply:
I do not create a separate Serializer now.